### DISCUSSION TITLE
how to add context and reference window like source insight

### DISCUSSION THREAD

**Discussion by stolen-programmer** [SpartanJ/ecode#117] [Q&A]
The reason i like source insight is that it has realtime context preview and reference map tree window. If ecode has, i will very happy and change to ecode.

**Comment by SpartanJ** [comment]
Hi! I'm not sure what you mean, the "context preview" should be what we call "minimap" in ecode (the quick view of the source on the right, I can see Source Insight have it). The tree map window I guess is the document symbol tree, ecode provides something similar but not as a tree, it's a symbol flat map, where you can fast fuzzy search you symbol with ctrl+shift+p and there's another global symbol search with ctrl+shift+k (you'll need a working LSP for all the symbol search capabilities). I might add a more traditional symbol tree later, but I don't personally use it since it's very mouse-driven and breaks a fast coding flow.
